src/sentry/api/serializers/models/user.py

```diff
@@ -174,10 +185,11 @@ def serialize(
             ],
         }
 
-        if obj == user:
+        if self._user_is_requester(obj, user):
```

Comment:
It will unfortunately be the case now that the `APIUser` and `User` objects will mix in some control silo code, depending on the context.  We will need to increasingly be careful of this.